### PR TITLE
Add org.gpodder.gpodder-adaptive

### PR DESCRIPTION
**Pull request by tpikonen**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)* (see below)
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

The adaptive version of `org.gpodder.gpodder`, maintained at https://github.com/gpodder/gpodder/tree/adaptive

Extra permissions:
- Needs access read-write access to homedir to manage downloaded media files
- Needs to own `org.gpodder.gpodder` name in order to prevent other gPodder versions running concurrently on the same database

### CODE REVIEW

File: org.gpodder.gpodder-adaptive.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  "runtime-version": "42",
+  "sdk": "org.gnome.Sdk",
+  "command": "gpodder",
+  "rename-icon": "gpodder-adaptive",
+  "copy-icon": true,
+  "rename-desktop-file": "gpodder.desktop",
+  "finish-args": [
+    "--device=dri",
+    "--socket=x11",
+    "--socket=wayland",
```
Comment (hfiguiere) [inline org.gpodder.gpodder-adaptive.json:13]:
```suggestion
    "--socket=fallback-x11",
    "--socket=wayland",
```

it it really supports wayland.

File: org.gpodder.gpodder-adaptive.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    "--share=ipc",
+    "--share=network",
+    "--own-name=org.gpodder.*",
```
Comment (hfiguiere) [inline org.gpodder.gpodder-adaptive.json:16]: `org.gpodder.gpodder-adaptive` is granted automatically. Why do you need this

File: org.gpodder.gpodder-adaptive.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    "--talk-name=org.gtk.vfs.*",
+    "--filesystem=home"
```
Comment (hfiguiere) [inline org.gpodder.gpodder-adaptive.json:18]: a select number of xdg- locations should be enough.

File: org.gpodder.gpodder-adaptive.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  ],
+  "cleanup": [
+    "/include",
+    "/lib/debug",
```
Comment (hfiguiere) [inline org.gpodder.gpodder-adaptive.json:22]:
```suggestion
```

File: org.gpodder.gpodder-adaptive.json [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      "name": "gpodder-adaptive",
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://github.com/gpodder/gpodder/archive/refs/tags/adaptive/3.11.0+1.tar.gz",
+          "sha256": "f30aa3a477ffecf7cfc4b65913902556499d3710127cd10d12ff46a70321320b"
+        },
+        {
+          "type": "file",
+          "path": "appdata.xml"
```
Comment (hfiguiere) [inline org.gpodder.gpodder-adaptive.json:38]: at has an appdata upstream. 

 https://github.com/gpodder/gpodder/blob/master/share/metainfo/org.gpodder.gpodder.appdata.xml

you shoudln't have one here

File: org.gpodder.gpodder-adaptive.json [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        }
+      ],
+      "buildsystem": "simple",
+      "build-commands": [
+        "make install PREFIX=$FLATPAK_DEST",
+        "mv $FLATPAK_DEST/share/applications/gpodder-url-handler.desktop $FLATPAK_DEST/share/applications/$FLATPAK_ID.gpodder-url-handler.desktop"
+      ],
+      "post-install": [
```
Comment (hfiguiere) [inline org.gpodder.gpodder-adaptive.json:46]:
```suggestion
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]
not sure about same app 

https://github.com/flathub/org.gpodder.gpodder

**Comment by hfiguiere** [comment]
> not sure about same app
> 
> https://github.com/flathub/org.gpodder.gpodder

To clarify, since they are from the same source, they shouldn't be duplicated.

**Comment by hfiguiere** [comment]
bot, build org.gpodder.gpodder-adaptive

**Comment by tpikonen** [comment]
Thanks for the review, I'll get back to it later.

> To clarify, since they are from the same source, they shouldn't be duplicated.

This branch of gPodder is usable on mobile phones (and desktop alike) with libhandy UI, while the other package is the classic WIMP experience. The org.gpodder.gpodder package is not usable on small touchscreens, but both branches have their users.

File: org.gpodder.gpodder-adaptive.json (hunk 2 above)
Comment (tpikonen) [reply org.gpodder.gpodder-adaptive.json:16]: This is to make `org.gpodder.gpodder` and this package co-installable, but not able to start at the same time (both versions claim org.gpodder.gpodder as application-id).

I'm not sure if `--own-name=org.gpodder.gpodder` would be enough though.

Comment (hfiguiere) [reply org.gpodder.gpodder-adaptive.json:16]: hence what I said about being the same app.

**Comment by hfiguiere** [comment]
```
builddir/files/share/appdata/org.gpodder.gpodder-adaptive.appdata.xml: FAILED:
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://raw.githubusercontent.com/flathub/org.gpodder.gpodder-adaptive/master/screenshot1.png]
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://raw.githubusercontent.com/flathub/org.gpodder.gpodder-adaptive/master/screenshot2.png]
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://raw.githubusercontent.com/flathub/org.gpodder.gpodder-adaptive/master/screenshot4.png]
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://raw.githubusercontent.com/flathub/org.gpodder.gpodder-adaptive/master/screenshot3.png]
Validation of files failed
```

**Comment by nedrichards** [comment]
What's the future expectation here? If the adaptive version expected to replace the 'classic' one? Will they continue in parallel indefinitely?

File: org.gpodder.gpodder-adaptive.json (hunk 3 above)
Comment (tpikonen) [reply org.gpodder.gpodder-adaptive.json:18]: gPodder is a program which deals with media files downloaded from feeds. The default location for the downloaded files is  `~/gPodder/Downloads`, which is also used by the org.gpodder.gpodder package (it has the `--filesystem=home` arg). Having a different location for downloaded files would be really inconvenient for people who want test this version with their existing installation.

File: org.gpodder.gpodder-adaptive.json (hunk 5 above)
Comment (tpikonen) [reply org.gpodder.gpodder-adaptive.json:37]: True, the upstream appdata is outdated though, so fixing that requires a new release.

**Comment by tpikonen** [comment]
> What's the future expectation here? If the adaptive version expected to replace the 'classic' one? Will they continue in parallel indefinitely?

There are no concrete plans to merge or replace these versions in the near future. The maintainers seem to like the classic 
interface.

File: org.gpodder.gpodder-adaptive.json (hunk 5 above)
Comment (hfiguiere) [reply org.gpodder.gpodder-adaptive.json:37]: or a patch.

File: org.gpodder.gpodder-adaptive.json (hunk 3 above)
Comment (hfiguiere) [reply org.gpodder.gpodder-adaptive.json:18]: then explictely allow that folder.

**Comment by nedrichards** [comment]
bot, build org.gpodder.gpodder-adaptive

**Comment by nedrichards** [comment]
Sure. It's unusual in a case like this for something to just sit in a long lived branch and not become a 'proper' fork. Certainly it'll be tough to keep them both going in the same repository with the same quality as time goes on.

**Comment by tpikonen** [comment]
> Certainly it'll be tough to keep them both going in the same repository with the same quality as time goes on.

It would be even tougher to maintain a proper fork without contributions from other maintainers.

**Comment by tpikonen** [comment]
Ok, the build _should_ work now. Thanks for the review so far.

**Comment by barthalion** [comment]
bot, build org.gpodder.gpodder-adaptive

**Comment by barthalion** [comment]
```
builddir/files/share/appdata/org.gpodder.gpodder-adaptive.appdata.xml: FAILED:
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://raw.githubusercontent.com/flathub/org.gpodder.gpodder-adaptive/master/screenshot1.png]
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://raw.githubusercontent.com/flathub/org.gpodder.gpodder-adaptive/master/screenshot2.png]
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://raw.githubusercontent.com/flathub/org.gpodder.gpodder-adaptive/master/screenshot4.png]
• url-not-found         : <screenshot> failed to download (HTTP 404: Not Found) [https://raw.githubusercontent.com/flathub/org.gpodder.gpodder-adaptive/master/screenshot3.png]
```

**Comment by tpikonen** [comment]
Thanks, the screenshots are now pointing to my branch.

**Comment by barthalion** [comment]
bot, build org.gpodder.gpodder-adaptive

**Pull request closed without merging**